lib/sensors/sensors.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+
+namespace hyped::sensors {
+
+enum class WriteResult { kFailure, kSuccess };
+
+/**
+ * If a sensor is to be used with an I2C mux, it must inherit from this abstract class.
+ */
+class II2cSensor {
+ public:
+  /*
+   * This function carries out the initilization steps for a particular sensor.
+   */
+  virtual bool configure()    = 0;
+  virtual WriteResult write() = 0;
+  virtual uint16_t read()     = 0;
```

Comment:
https://github.com/Hyp-ed/hyped-2023/blob/9a3285262877e8cb0e3a3bdb00dadacc28cbcf76/lib/io/i2c.cpp#L43

Would we want the sensor returning an optional since the i2c interface does?

Reply:
Good point, will change